runtime/src/node_modules/@sapper/internal/App.svelte.d.ts

```diff
@@ -0,0 +1,8 @@
+import {
+  ComponentConstructor,
+  SSRComponent
+} from './shared';
+
+declare const App: ComponentConstructor & SSRComponent
```

Comment:
This throws type errors for me when trying it in the TS playground, because the generic for `SSRComponent` is missing.

To get the generics in there and have it types as a class at the same time without writing everything twice, you can do class/interface merging:

```ts
declare interface App<Props> extends ComponentConstructor, SSRComponent<Props> {}
declare class App<Props> {}
```

Reply:
I just installed VS Code and it led me down a rabbit hole of redoing most of the types. It showed a bunch of errors that the command line build wasn't. Is VS Code doing something different and more powerful than the rollup plugin does?

Reply:
Ok, I managed to take the files out of `node_modules`, which is making things work better for me